File: DataStructures/src/main/java/org/mainProgramme/SortedLinkedList.java
package org.mainProgramme;

import org.mainProgramme.Node.SortedNode;

public class SortedLinkedList<T extends Comparable<T>> {
    public SortedNode<T> head = null;
    public SortedNode<T> tail = null;

    public void add(T value) {
        SortedNode<T> newNode = new SortedNode<>(value);
        SortedNode<T> current = head;
        if (head == null) {
            head = newNode;
            tail = newNode;
        }
        else if (newNode.value.compareTo(head.value) <= 0) {
            newNode.next = head;
            head.previous = newNode;
            head = newNode;
        }
        else if (newNode.value.compareTo(tail.value) >= 0) {
            newNode.previous = tail;
            tail.next = newNode;
            tail = newNode;
        }
        else {
            while (current != null) {
                if (newNode.value.compareTo(current.value) > 0) {
                    current = current.next;
                    if (newNode.value.compareTo(current.value) < 0) {
                        newNode.next = current;
                        newNode.previous = current.previous;
                        current.previous.next = newNode;
                        current.previous = newNode;
                    }
                }
            }
        }
    }

    public void displayValue() {
        SortedNode<T> current = head;
        while (current != null) {
            System.out.println(current.value);
            current = current.next;
        }
    }
}

File: LinkedList/LinkedList/SortedLinkedList.py
from Node.SortedNode import sortedNode
class sortedLinkedList:
    def __init__(self):
        self.head = None
        self.tail = None

    def add(self, value):
        newNode = sortedNode(value)
        current = self.head
        if self.head is None:
            self.head = newNode
            self.tail = newNode
        elif newNode.__lt__(self.head.value) <= 0:
            newNode.next = self.head
            self.head.previous = newNode
            self.head = newNode
        elif newNode.__lt__(self.tail.value) >= 0:
            newNode.previous = self.tail
            self.tail.next = newNode
            self.tail = newNode
        else:
            while newNode.__lt__(current.value) > 0:
                current = current.next
            current.previous.next = newNode
            newNode.previous = current.previous
            newNode.next = current
            current.previous = newNode

    def display(self):
        current = self.head
        while current != None:
            print(current.value)
            current = current.next
